lib/rules/valid-api.js

```diff
@@ -88,6 +93,13 @@ function validatePropertyName(property, context) {
             message,
         });
     }
+
+    if (name.match(MIX_UPPERCASE_AND_UNDERSCORE)) {
```

Comment:
As discussed here https://github.com/salesforce/eslint-plugin-lwc/pull/54#discussion_r534083307, it would be better to have this feature behind a flag to avoid the breaking change.

Reply:
Because we semver this package, a breaking change in behavior is fine if it's coupled with a semver bump. sfdx needs to bump the version of this package based on org API version as we discussed yesterday. 

If this rule is enforced at DB save time then it needs to be tied to org API and/or module API version, which requires a longer discussion.

Reply:
This rule is enforced at save time. Since the LWC compiler doesn't have yet access to the API version at compile-time, we can't land a breaking change right now.